pkg/collector/corechecks/snmp/devicecheck/devicecheck.go

```diff
@@ -0,0 +1,160 @@
+package devicecheck
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/checkconfig"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/common"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/fetch"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/metadata"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/report"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/session"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/valuestore"
+)
+
+const (
+	snmpLoaderTag    = "loader:core"
+	serviceCheckName = "snmp.can_check"
+)
+
+// DeviceCheck hold info necessary to collect info for a single device
+type DeviceCheck struct {
+	config  *checkconfig.CheckConfig
+	sender  *report.MetricSender
+	session session.Session
+}
+
+// NewDeviceCheck returns a new DeviceCheck
+func NewDeviceCheck(config *checkconfig.CheckConfig, ipAddress string) (*DeviceCheck, error) {
+	newConfig := config.CopyWithNewIP(ipAddress)
+
+	sess, err := session.NewSession(newConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to configure session: %s", err)
+	}
+
+	return &DeviceCheck{
+		config:  newConfig,
+		session: sess,
+	}, nil
+}
+
+// SetSender sets the current sender
+func (d *DeviceCheck) SetSender(sender *report.MetricSender) {
```

Comment:
As we have `metadata` and `metrics` senders, should this be more specific, like `SetMetricSender` ?

Reply:
I see the sender is used for both metrics and metadata, could `MetricSender` type be just `Sender` ?

Reply:
Yes, possible, called it `Sender` might be confusing with the `aggregator.Sender`, maybe `SnmpSender` or `CustomSender` ?

Do you mind if I extract this to a separate future PR ? Since it can lead to renaming in a lot of places, I might makes the https://github.com/DataDog/datadog-agent/pull/8996 PR a bit difficult to rebase on this one.